components/director/internal/destinationfetchersvc/service.go

```diff
@@ -99,57 +101,61 @@ func (d *DestinationService) generateClientBySubdomainLabel(ctx context.Context,
 		return nil, errors.New("cannot cast label value as a string")
 	}
 
 	instanceConfig, ok := d.DestinationsConfig.RegionToInstanceConfig[region]
 	if !ok {
-		log.C(ctx).Errorf("No destination instance credentials found for region '%s'", region)
-		return nil, errors.New(fmt.Sprintf("No destination instance credentials found for region '%s'", region))
+		return nil, fmt.Errorf("no destination instance credentials found for region '%s'", region)
 	}
 
 	client, err := NewClient(instanceConfig, d.APIConfig, subdomain)
 	if err != nil {
-		log.C(ctx).WithError(err).Error("Failed to create Destination API client")
-		return nil, err
+		return nil, errors.Wrap(err, "failed to create Destination API client")
 	}
 
 	return client, nil
 }
 
 // SyncTenantDestinations syncs destinations for a given tenant
 func (d *DestinationService) SyncTenantDestinations(ctx context.Context, tenantID string) error {
+	log.C(ctx).Infof("Starting sync of destinations for tenant '%s'", tenantID)
+
 	subdomainLabel, err := d.getSubscribedSubdomainLabel(ctx, tenantID)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "failed to get subdomain label for tenant '%s'", tenantID)
 	}
 
 	client, err := d.generateClientBySubdomainLabel(ctx, subdomainLabel)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "failed to create destinations client for tenant '%s'", tenantID)
 	}
 
+	// Make it debugf
+	log.C(ctx).Infof("Syncing destinations for tenant '%s' with subdomain '%s'", tenantID, subdomainLabel.Value)
+
 	revision := d.UUIDSvc.Generate()
-	err = d.walkthroughPages(ctx, client, func(destinations []destinationFromService) error {
-		log.C(ctx).Infof("Found %d destinations in tenant '%s'", len(destinations), tenantID)
-		return d.mapDestinationsToTenant(ctx, *subdomainLabel.Tenant, revision, destinations)
+	err = d.walkthroughPages(ctx, client, tenantID, func(destinations []destinationFromService) error {
+		start := time.Now()
+		err := d.mapDestinationsToTenant(ctx, tenantID, revision, destinations)
+		log.C(ctx).Infof("Mapping destinations for tenant '%s' took %s", tenantID, time.Since(start)) // remove
+		return err
 	})
 	if err != nil {
-		log.C(ctx).WithError(err).Errorf("Failed to sync destinations for tenant '%s'", tenantID)
-		return err
+		return errors.Wrapf(err, "failed to sync destinations for tenant '%s'", tenantID)
 	}
+	// Make it debugf
+	log.C(ctx).Infof("Upserted destinations for tenant '%s'. Removing old destinations...", tenantID)
 
-	if err := d.deleteMissingDestinations(ctx, revision, *subdomainLabel.Tenant); err != nil {
-		log.C(ctx).WithError(err).Errorf("Failed to delete missing destinations for tenant '%s'", tenantID)
-		return err
+	if err := d.deleteMissingDestinations(ctx, revision, tenantID); err != nil {
+		return errors.Wrapf(err, "failed to delete missing destinations for tenant '%s'", tenantID)
 	}
+	log.C(ctx).Infof("Finished sync of destinations for tenant '%s'", tenantID)
```

Comment:
We could also print the execution time here, so we can see it in the case of force reload